### ISSUE TITLE
"command insert requires authentication", db adapter not production-ready?

### ISSUE THREAD

**Issue by tvld** [feathersjs-ecosystem/feathers-mongoose#389]
New in Feathersjs this week, we generated a server and tried to POST to ``/users`` but response is:
```json
{
  "name": "GeneralError",
  "message": "command insert requires authentication",
  "code": 500,
}
```
However, the connection work when I fire up the mongo docker _without_ the username passwords, disabling 'auth' (!).
```conf
version: "3"

services:
  mongo:
    image: mongo
    restart: always
    environment:
      MONGO_INITDB_DATABASE: "my_database"
      # MONGO_INITDB_ROOT_USERNAME: "admin_user"
      # MONGO_INITDB_ROOT_PASSWORD: "password123"
    ports:
      - "27017:27017"
    volumes:
      # - /opt/mongo/db:/data/db
      - ./initdb:/docker-entrypoint-initdb.d:ro
```
Obvisouly I tried to enter the user / name password to authorise mongodb, and tried in the config/default.json add some fields... which it did not pick up. Puzzled, I see no where in the docs I see how to make this a secure set up and enter the username/ password to auth:

- is it safe to assume,  _this db adapter is not yet production- ready_?
- are there plans to add this basic security?
- would it help if we add some PR? (although being fully new on this, it would take time and courage;))))

**Comment by rnbrady** [comment]
Did you try setting the username and password in `mongoose.ts`?

```typescript
export default function (app: Application) {
  mongoose.connect(
    app.get('mongodb'),
    { 
      user: "admin_user",
      pass: "password123"
      ...
```

**Comment by daffl** [comment]
After 6 years, 65+ releases and almost a million downloads I don't think production readiness is the problem here. All settings from the [Mongoose connection documentation](https://mongoosejs.com/docs/connections.html) should work as described. If not, first verify if using Mongoose directly is causing the same problem. If it does not, please provide an example repository, Docker container configuration and steps to reproduce the issue.

**Comment by tvld** [comment]
Thank you for the swift response. @rnbrady answers works, but one should  include ``authSource=admin`` at the end of the url. I decided to use simply:
```js
"mongodb": "mongodb://admin_user:password123@localhost:27017/my_database?authSource=admin"
```
I was a bit confused why there was no feathers info on hardening the install. But I am now getting that this is all for the database to handle. A week further now and glad we took the jump to catch a feather ;-)  Thanks guys !!!!
